src/util.ts

```diff
@@ -12,7 +12,60 @@
  * language governing permissions and limitations under the License.
  */
 
+import {IonTypes} from "./IonTypes";
+import {IonType} from "./IonType";
+import {Reader as IonReader} from "./IonReader";
+import {Writer as IonWriter} from "./IonWriter";
+
+/**
+ * Returns -1 if x is negative (including -0); otherwise returns 1.
+ */
 export function _sign(x: number): number {
     return (x < 0 || (x === 0 && (1 / x) === -Infinity)) ? -1 : 1
 }
 
+/**
+ * Writes a reader's current value and all following values until the end
+ * of the current container.  If there's no current value then this method
+ * calls Reader.next() to get started.
+ */
+export function _writeValues(reader: IonReader, writer: IonWriter, _depth = 0): void {
+    let type: IonType = reader.type();
+    if (type === null) {
+        type = reader.next();
+    }
+    while (type !== null) {
+        if (_depth > 0) {
```

Comment:
But there is one thing:  does this mean we will not write field names if the user has manually walked the reader into a struct?

Reply:
that's correct;  see the "_writeValues(), start within container" test